src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -3985,7 +3985,13 @@ module EstablishTypeDefinitionCores =
                 match fields' with 
                 | rf :: _ -> errorR (Error(FSComp.SR.tcInterfaceTypesAndDelegatesCannotContainFields(), rf.Range))
                 | _ -> ()
-
+            
+            let primaryConstructorInDelegateCheck(implicitCtorSynPats : SynSimplePats option) =
+                match implicitCtorSynPats with 
+                | None -> ()
+                | Some spats ->
+                    let ctorArgNames, _ = TcSimplePatsOfUnknownType cenv true CheckCxs envinner tpenv spats
+                    if not ctorArgNames.IsEmpty then errorR (Error(FSComp.SR.parsOnlyClassCanTakeValueArguments(), m))
```

Comment:
I wonder, what is inside `ctorArgNames` when there is an empty constructor?

Reply:
```fsharp
namespace FSharpTest
    type T() =
        delegate of int -> int
```

<img width="500" alt="Capture" src="https://user-images.githubusercontent.com/31915729/171026804-8c588bf6-1f8b-4200-85b2-ec673b1c1546.PNG">